### ISSUE TITLE
Antenna animated target tracking, like in original RT1?

### ISSUE THREAD

**Issue by StoneBlue** [RemoteTechnologiesGroup/RemoteTech#705]
In the original version of RemoteTech, there was a beautiful animated, double-jointed dish that could physically track a connected target, similar to what solar panels do with tracking the sun.
From what I recall, an update to Unity/KSP broke the ability to do this for target tracking.

With several Unity/KSP updates since then, would it be possible to revisit whether this can be implemented again? 
If it can, it could open up new possibilities for a slew of new antenna designs/mods.

Thanx!

**Comment by neitsa** [comment]
Thank you @StoneBlue for your request!

Do you happen to remember which one of the [antennas ](http://remotetechnologiesgroup.github.io/RemoteTech/guide/parts/#dish-antennas )it was? The [GX-128](http://remotetechnologiesgroup.github.io/RemoteTech/guide/parts/#reflectron-gx-128) maybe?

**Comment by StoneBlue** [comment]
no, not the GX-128... WAY, way earlier... It was the MacroDish.
heres a post with a pic. its the antenna on the far right, bottom pic:
http://forum.kerbalspaceprogram.com/index.php?/topic/16257-020-remotetech-relay-network-%C3%A3%C2%A2%C3%A2%E2%80%9A%C2%AC%C3%A2%E2%82%AC%C5%93-v-0501/&do=findComment&comment=458748

I dug around the thread, but couldnt find the discusiion about why they had to drop the "mechanics" for it... But it had to have been sometime between this post, and the 0.21 update, or the release of RT2...

Then there's also this OLD, original RT2 thread that it might be in:
http://forum.kerbalspaceprogram.com/index.php?/topic/51552-now-defunct-thread-that-should-not-appear-in-google-search/&page=1

I know someone released ALL the original RT parts (tracking antenna/aeroprobe/etc that were dropped from the original RT), and they were updated to work with KSP 0.90. I have a copy of it, and I tried getting the parts working in 1.1.x... most do, with a few weird things I couldnt figure out... The MacroDish still works, animations and all, EXCEPT for the target tracking... Which I knew wouldnt work, since the module is so old....

Heres a few screenshots of some of the parts, IIRC, in KSP 1.1.x:

https://imgur.com/a/96ttW

**Comment by YamoriYuki** [comment]
@StoneBlue We do not have access to those models or code so we need to make sure that including them in RemoteTech is OK. Either by finding a trustworthy source that includes license permitting such reuse or contacting the author. My attempt at it failed miserably though. Would it be ok to ask you to help us out with that?

**Comment by JDPKSP** [comment]
Just dropping by with some historical info of a bygone age. Target tracking dishes where a feature of the original RT1 mod in the days where I was the sole developer. When Cilph took over as main developer and basically rewrote the entire plugin, tracking dishes where put on hold for eventual later reimplementation. Since tracking dishes would need to be made with animated joints in mind and neither Cilph nor myself where much good with modeling it was simply never reimplemented since we had to mainly use models made by other modders.
For instance, take a look at the GX-128, which is the last remnant of my old models. I still hold that the folding animation is pretty cool. But the modeling and texturing leave much to be desired.
I think I only ever made one tracking dish, I can't recall what I ended up naming it. Anyways [here](http://www.youtube.com/watch?v=AhPUuGFrA4s) is a dev video of it in action. I might have the files on a drive somewhere. Of course anyone is free to use, modify and redistribute any model I've made as they see fit (unless you´re gonna sell it, but I highly doubt you'd find any buyers).
The MacroDish was made by Spaceghöst and was made specifically to use the tracking functionality. Therefore I can't really give any permissions with regard to that model, You'd have to find and contact Spaceghöst.
If anyone wants to take a look at the original module for tracking antennae, it's available through the RT1 legacy repo [here](https://github.com/JDPKSP/RemoteTechLegacy/blob/master/ModuleRTAnimatedTrackAntenna.cs)

That being said. If tracking dishes make their return to RemoteTech, new models should be made to keep with the stock look. It does seem like most of the current models lend themselves well to be replaced with a slightly modified version to provide the needed joints. If that is ever done, it could be prudent to revamp the GX-128 to get rid of the last remnant of the old minimalist RT1 look. Which was mostly a product of my own artistic shortcomings.

**Comment by sparr** [comment]
Would you expect a dish to not work if the target is outside its tracking range of motion?

**Comment by JDPKSP** [comment]
Originally the physical tracking behavior was completely distinct from actual targeting. The physical tracking animation was purely visual.
IIRC, the main reason for this was that requiring dishes to be able to actually track their target within their signal cones would introduce way too much obscure complexity into gameplay. Leading to intermittent loss of signal within the network that to the player might seem random.
Of course there would also be the issue of how to handle dish tracking on unloaded vessels, which - while doable, at the cost of more computation in link evaluation - would just introduce more sources for obscure loss of signal events.

**Comment by sparr** [comment]
I'd be happy to have unloaded vessels assume they stay oriented for comms. This mirrors my wish that I could get vessels to stay oriented for solar...

**Comment by JDPKSP** [comment]
Just be aware that staying oriented is a trivial problem when you are only referring to a single target, but when you introduce multiple, you will most likely end up in situations where you cannot maintain tracking for all targets simultaneously. In that case, the satellite would need to periodically reorient for different targets to be able to function as a relay satellite. Continuously caching incoming signals before reorienting to rebroadcast them.

Almost exactly the same results can be reproduced simply by assuming that any dish can track any target, but at a much lower computational cost and without introducing unpredictable delays due to signal caching.
